packages/framework/aqueduct/src/data-object-factories/migrationDataObjectFactory.ts

```diff
@@ -119,7 +120,8 @@ export interface MigrationDataObjectFactoryProps<
  * @alpha
  */
 export class MigrationDataObjectFactory<
```

Comment:
So far we'd still need different `SharedFluidLoopComponentFactory` variants in the Loop SDK repo to account for MigrationDataObject and DataObject, right?

Reply:
My vision is that we'll only have MigrationDataObject, but with a different name like FlexibleDataObject (Not a real suggestion but gets the point across).  Then insteade of TreeDataObject, you'd make a FlexibleDataObject with just one ModelDescriptor.  This way in the future it's ready to migrate if a component wants to make a drastic change (like folding SharedString into the Tree)

Reply:
*Maybe* we could even put this all on DataObject and slowly deprecate/remove `root` channel (replace it with a default Model Descriptor)

Reply:
Hm... that's a good one to discuss. So far, I've been thinking that we'd want TreeDataObject for new components, because I was thinking we can avoid the complexity of migration overhead in this case. Having said that, I think you're right that even in such a case there might be reasons for wanting a data migration further down the line, and not having to change base classes at that point would be a benefit. If we're doing a good job in API design, the migration overhead is kept at a minimum too.